pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,264 @@
+      cp -r include/* $out/include
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    # Tied to the tagged version of EDOPro's git module.
+    version = "4a872ad7686e0d00ee4c1c051e90fc07c40613d4";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = version;
+      hash = "sha256-2E1cjG0FONu/fbr67/3qRCKQ1W/wPznClEWsMa1FAzo=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      lua5_3
+    ];
+
+    preBuild = ''
+      premake5 gmake2
+    '';
+
+    enableParallelBuilding = true;
+    buildFlags = "verbose=true config=release_x64 ocgcore";
+    makeFlags = "-C build";
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/libocgcore.a $out/bin
+    '';
+  };
+
+  edopro = stdenv.mkDerivation rec {
+    pname = "edopro";
+    version = edopro-version;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = "c713e23491a1e55c9d8e91257e5f2b5873696b9b";
+      hash = "sha256-2E1cjG0FONu/fbr67/3qRCKQ1W/wPznClEWsMa1FAzo=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      bzip2
+      curl
+      flac
+      fmt_9
+      freetype
+      irrlicht-edopro
+      libevent
+      libgit2
+      libjpeg
+      libpng
+      libvorbis
+      mono
+      nlohmann_json
+      openal
+      SDL2
+      sqlite
+    ];
+
+    # nixpkgs' gcc stack currently appears to not support LTO
+    postPatch = ''
+      sed -i '/LinkTimeOptimization/d' ./premake5.lua
+    '';
+
+    # TODO sound, can't get it to work with either SDL2 or sfml
+    preBuild = ''
+      premake5 gmake2 \
+        --bundled-font="${font}" \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --no-core \
+        --no-joystick=false \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+    '';
+
+    enableParallelBuilding = true;
+    NIX_CFLAGS_COMPILE = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    buildFlags = "verbose=true config=release_x64 ygoprodll";
+    makeFlags = "-C build";
+
+    # TODO find out how to get arch from nix for cross compiled
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        #"cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
+      set -eu
+      EDOPRO_DIR="$HOME/.local/share/edopro"
+
+      if [ ! -d $EDOPRO_DIR ]; then
+          mkdir -p $EDOPRO_DIR
+          cp -r ${assets.src}/{${assetsToCopy}} $EDOPRO_DIR
```

Comment:
This isn't quite ready for general use, ideally this should have some logic that prevents overwriting user edits but recovers the directory if it's broken. Also applying updates would be difficult with the current script, and will probably become important eventually.

Reply:
You wrote this code, I've made no changes

Reply:
Yeah, which is why I point it out as a remaining problem ;p

Reply:
I'll see if I can think of a better way to do this sometime this week, don't have *that* much disposable evening time currently, though.

Reply:
I asked edopro about this and he confirmed the files we need to never override if they exist they are.
* config/user_configs.json
* deck
* pics
* replays

Everything else we can nuke on an update.